### PR TITLE
Remove SwiftASTContext from HashedCollectionConfig::StorageObjectAtAddress()

### PR DESCRIPTION
**Pull request by adrian-prantl**
rdar://88509698

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/Language/Swift/SwiftDictionary.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ DictionaryConfig::DictionaryConfig()
   // still used to name classes in the ObjC runtime.
 
   m_nativeStorageRoot_mangled =
-    ConstString("_TtCs22__RawDictionaryStorage");
+    ConstString("$ss22__RawDictionaryStorageCD");
```
Comment (adrian-prantl) [inline lldb/source/Plugins/Language/Swift/SwiftDictionary.cpp:48]: This is necessary because TypeSystemSwiftTypeRef only accepts the new mangling.

Comment (kastiglione) [reply lldb/source/Plugins/Language/Swift/SwiftDictionary.cpp:48]: Which part of TypeSystemSwiftTypeRef is that?

Comment (adrian-prantl) [reply lldb/source/Plugins/Language/Swift/SwiftDictionary.cpp:48]: The Verify() function

Comment (adrian-prantl) [reply lldb/source/Plugins/Language/Swift/SwiftDictionary.cpp:48]: Sorry that is incorrect. The problem is
```
swift::Type SwiftASTContext::GetSwiftType(opaque_compiler_type_t opaque_type) {
  assert(opaque_type && *reinterpret_cast<const char *>(opaque_type) != '$' &&
         "wrong type system");
```
which triggers in the VALIDATION macros when lifting the type into the SwiftASTContext type system.

**Pull request merged**